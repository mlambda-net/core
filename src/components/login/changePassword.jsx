import React from 'react';
import { Box, Button, FormControl, Paper, TextField } from '@mui/material';
import LocalizedStrings from 'react-localization';
import { withUtils } from '@mlambda-net/web-core/utils';
import { Title } from '@mlambda-net/web-core/common';

const language = new LocalizedStrings({
  en: {
    title: 'Change Password',
    newPassword: 'New Password',
    oldPassword: 'Old Password',
    email: 'Email',
    errorNewPassword: 'new password is required',
    errorOldPassword: 'old password is required',
    changeAction: 'submit',
  },
  es: {
    title: 'Cambiar contraseña',
    newPassword: 'Contraseña Nueva',
    oldPassword: 'Contraseña Actual',
    email: 'Email',
    errorNewPassword: 'la contraseña nueva es requerida',
    errorOldPassword: 'la contraseña actual es requerida',
    changeAction: 'cambiar',
  },
});

const styles = (theme) => ({});

class ChangePassword extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      newPassword: '',
      oldPassword: '',
      validOldPassword: false,
      validNewPassword: false,
      errorNewPassword: '',
      errorOldPassword: '',
    };

    this.valid = this.props.validation;
    this.newPasswordChange = this.newPasswordHandle.bind(this);
    this.oldPasswordChange = this.oldPasswordHandle.bind(this);
    this.changeClick = this.changeHandle.bind(this);
  }

  render() {
    language.setLanguage(this.props.lang);
    const { classes } = this.props;
    return (
      <Paper elevation={10} className={this.props.className}>
        <Box p={3}>
          <Box display="flex" justifyContent="center">
            <Title title={language.title} />
          </Box>
          <Box
            display="flex"
            flexDirection="column"
            justifyContent="center"
            m={1}
            p={1}>
            <Box p={1}>
              <FormControl fullWidth>
                <TextField
                  id="password"
                  label={language.oldPassword}
                  color="secondary"
                  type="password"
                  error={!this.state.validOldPassword}
                  onChange={this.oldPasswordChange}
                  helperText={this.state.errorOldPassword}
                />
              </FormControl>
            </Box>

            <Box p={1}>
              <FormControl fullWidth>
                <TextField
                  id="confirm_password"
                  label={language.newPassword}
                  color="secondary"
                  type="password"
                  error={!this.state.validNewPassword}
                  onChange={this.newPasswordChange}
                  helperText={this.state.errorNewPassword}
                />
              </FormControl>
            </Box>

            <Box p={1} m={1} display="flex" justifyContent="flex-end">
              <Button
                variant="outlined"
                color="primary"
                disabled={!this.isValid()}
                onClick={this.changeClick}>
                {language.changeAction}
              </Button>
            </Box>
          </Box>
        </Box>
      </Paper>
    );
  }

  newPasswordHandle(event) {
    const password = event.target.value;
    if (this.valid.isEmpty(password)) {
      this.setState({
        validOldPassword: false,
        errorNewPassword: language.errorNewPassword,
      });
    } else {
      this.setState({
        newPassword: password,
        validNewPassword: true,
        errorNewPassword: '',
      });
    }
  }

  oldPasswordHandle(event) {
    const password = event.target.value;
    if (this.valid.isEmpty(password)) {
      this.setState({
        validOldPassword: false,
        errorOldPassword: language.errorOldPassword,
      });
    } else {
      this.setState({
        oldPassword: password,
        validOldPassword: true,
        errorOldPassword: '',
      });
    }
  }

  changeHandle(event) {
    this.props.onChange({
      newPassword: this.state.newPassword,
      oldPassword: this.state.oldPassword,
    });

    event.preventDefault();
  }

  isValid() {
    return this.state.validNewPassword && this.state.validOldPassword;
  }
}

export default withUtils(styles)(ChangePassword);
