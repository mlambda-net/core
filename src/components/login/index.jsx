export { default as Register } from './register';
export { default as SignIn } from './signin';
export { default as ChangePassword } from './changePassword';
export { default as Recovery } from './recovery';
export { default as Profile } from './profile';
