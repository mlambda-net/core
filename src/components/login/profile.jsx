import React from 'react';
import LocalizedStrings from 'react-localization';
import AccountCircleIcon from '@mui/icons-material/AccountCircle';
import { Box, Card, CardActions, CardContent, Typography } from '@mui/material';
import { Option } from '@mlambda-net/web-core/common';
import { withUtils } from '@mlambda-net/web-core/utils';

const language = new LocalizedStrings({
  en: {
    title: 'Profile',
  },
  es: {
    title: 'Perfil',
  },
});

const styles = (theme) => ({
  panel: {
    width: '100%',
    height: '100%',
  },

  title: {
    textAlign: 'center',
  },
  content: {
    height: 'calc(100% - 80px)',
  },
  actions: {
    display: 'flex',
    flexWrap: 'nowrap',
    justifyContent: 'flex-end',
  },
});

class Profile extends React.Component {
  constructor(props) {
    super(props);
    this.valid = this.validation;
    this.state = { open: false };
  }

  render() {
    language.setLanguage(this.props.lang);
    const Content = this.props.content;
    const Actions = this.props.actions;
    const { classes } = this.props;

    return (
      <Option
        icon={AccountCircleIcon}
        width={this.props.width}
        height={this.props.height}
        style={this.props.style}>
        <Card className={classes.panel}>
          <CardContent className={classes.content}>
            <Typography
              className={classes.title}
              color="textSecondary"
              variant="h5"
              gutterBottom>
              {language.title}
            </Typography>
            <Content user={this.props.user} />
          </CardContent>

          <CardActions className={classes.actions}>
            <Box>
              <Actions />
            </Box>
          </CardActions>
        </Card>
      </Option>
    );
  }
}

export default withUtils(styles)(Profile);
