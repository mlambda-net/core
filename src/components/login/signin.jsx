import React from 'react';
import LocalizedStrings from 'react-localization';
import { withUtils } from '@mlambda-net/web-core/utils';
import {
  Box,
  Button,
  FormControl,
  Link,
  TextField,
  Typography,
} from '@mui/material';

const language = new LocalizedStrings({
  en: {
    password: 'Password',
    email: 'Email',
    invalidEmail: 'invalid email',
    invalidPassword: 'password is required',
    forgot: 'forgot password?',
    login: 'login',
  },
  es: {
    password: 'Contraseña',
    email: 'Correo',
    invalidEmail: 'correo invalido',
    invalidPassword: 'una contraseña es requerida',
    forgot: 'olvido la contraseña?',
    login: 'ingresar',
  },
});

const styles = (theme) => ({});

class SignIn extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      validEmail: false,
      validPassword: false,
      helpEmail: '',
      helpPassword: '',
    };
    this.valid = this.props.validation;
    this.emailChange = this.emailChangeHandle.bind(this);
    this.passwordChange = this.passwordChangeHandle.bind(this);
    this.loginClick = this.loginHandle.bind(this);
  }

  render() {
    language.setLanguage(this.props.lang);

    return (
      <Box width="100%">
        <Box display="flex" justifyContent="center">
          <Typography color="primary" variant="h4" component="h5" gutterBottom>
            {this.props.title}
          </Typography>
        </Box>
        <Box
          display="flex"
          flexDirection="column"
          justifyContent="center"
          m={1}
          p={1}>
          <Box p={1}>
            <FormControl fullWidth>
              <TextField
                label={language.email}
                color="secondary"
                id="email"
                type="email"
                error={!this.state.validEmail}
                onChange={this.emailChange}
                helperText={this.state.helpEmail}
              />
            </FormControl>
          </Box>
          <Box p={1}>
            <FormControl fullWidth>
              <TextField
                id="password"
                label={language.password}
                color="secondary"
                type="password"
                error={!this.state.validPassword}
                onChange={this.passwordChange}
                helperText={this.state.helpPassword}
              />
            </FormControl>
          </Box>

          <Box p={1} m={2} display="flex" justifyContent="flex-end">
            <Link color="secondary" onClick={this.props.onForgot}>
              <Typography variant="caption">{language.forgot}</Typography>
            </Link>
          </Box>

          <Box p={1} m={1} display="flex" justifyContent="flex-end">
            <Button
              variant="outlined"
              color="primary"
              disabled={!this.isValid()}
              onClick={this.loginClick}>
              {language.login}
            </Button>
          </Box>
        </Box>
      </Box>
    );
  }

  emailChangeHandle(event) {
    if (this.valid.email(event.target.value)) {
      this.setState({
        validEmail: true,
        helpEmail: '',
        email: event.target.value,
      });
    } else {
      this.setState({ validEmail: false, helpEmail: language.invalidEmail });
    }
  }

  passwordChangeHandle(event) {
    if (!this.valid.isEmpty(event.target.value)) {
      this.setState({
        validPassword: true,
        helpPassword: '',
        password: event.target.value,
      });
    } else {
      this.setState({
        validPassword: false,
        helpPassword: language.invalidPassword,
      });
    }
  }

  loginHandle(event) {
    this.props.onLogin({
      email: this.state.email,
      password: this.state.password,
    });
    event.preventDefault();
  }

  isValid() {
    return this.state.validEmail && this.state.validPassword;
  }
}

export default withUtils(styles)(SignIn);
