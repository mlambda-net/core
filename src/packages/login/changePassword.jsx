import React from 'react';
import Paper from '@material-ui/core/Paper';
import FormControl from '@material-ui/core/FormControl';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import LocalizedStrings from 'react-localization';
import { withUtils } from '@mlambda-net/core/utils';
import { Title } from '@mlambda-net/core/common';

const language = new LocalizedStrings({
  en: {
    title: 'Change Password',
    newPassword: 'New Password',
    oldPassword: 'Old Password',
    email: 'Email',
    errorNewPassword: 'new password is required',
    errorOldPassword: 'old password is required',
    changeAction: 'submit',
  },
  es: {
    title: 'Cambiar contraseña',
    newPassword: 'Contraseña Nueva',
    oldPassword: 'Contraseña Actual',
    email: 'Email',
    errorNewPassword: 'la contraseña nueva es requerida',
    errorOldPassword: 'la contraseña actual es requerida',
    changeAction: 'cambiar',
  },
});

const styles = (theme) => ({});

class ChangePassword extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      newPassword: '',
      oldPassword: '',
      validOldPassword: false,
      validNewPassword: false,
      errorNewPassword: '',
      errorOldPassword: '',
    };

    this.valid = this.props.validation;
    this.newPasswordChange = this.newPasswordHandle.bind(this);
    this.oldPasswordChange = this.oldPasswordHandle.bind(this);
    this.changeClick = this.changeHandle.bind(this);
  }

  render() {
    language.setLanguage(this.props.lang);
    const { classes } = this.props;
    return (
      <Paper elevation={10} className={this.props.className}>
        <Box p={3}>
          <Box display="flex" justifyContent="center">
            <Title title={language.title} />
          </Box>
          <Box
            display="flex"
            flexDirection="column"
            justifyContent="center"
            m={1}
            p={1}>
            <Box p={1}>
              <FormControl fullWidth>
                <TextField
                  id="password"
                  label={language.oldPassword}
                  color="secondary"
                  type="password"
                  error={!this.state.validOldPassword}
                  onChange={this.oldPasswordChange}
                  helperText={this.state.errorOldPassword}
                />
              </FormControl>
            </Box>

            <Box p={1}>
              <FormControl fullWidth>
                <TextField
                  id="confirm_password"
                  label={language.newPassword}
                  color="secondary"
                  type="password"
                  error={!this.state.validNewPassword}
                  onChange={this.newPasswordChange}
                  helperText={this.state.errorNewPassword}
                />
              </FormControl>
            </Box>

            <Box p={1} m={1} display="flex" justifyContent="flex-end">
              <Button
                variant="outlined"
                color="primary"
                disabled={!this.isValid()}
                onClick={this.changeClick}>
                {language.changeAction}
              </Button>
            </Box>
          </Box>
        </Box>
      </Paper>
    );
  }

  newPasswordHandle(event) {
    const password = event.target.value;
    if (this.valid.isEmpty(password)) {
      this.setState({
        validOldPassword: false,
        errorNewPassword: language.errorNewPassword,
      });
    } else {
      this.setState({
        newPassword: password,
        validNewPassword: true,
        errorNewPassword: '',
      });
    }
  }

  oldPasswordHandle(event) {
    const password = event.target.value;
    if (this.valid.isEmpty(password)) {
      this.setState({
        validOldPassword: false,
        errorOldPassword: language.errorOldPassword,
      });
    } else {
      this.setState({
        oldPassword: password,
        validOldPassword: true,
        errorOldPassword: '',
      });
    }
  }

  changeHandle(event) {
    this.props.onChange({
      newPassword: this.state.newPassword,
      oldPassword: this.state.oldPassword,
    });

    event.preventDefault();
  }

  isValid() {
    return this.state.validNewPassword && this.state.validOldPassword;
  }
}

ChangePassword.protoTypes = {
  className: PropTypes.string,
  classes: PropTypes.object.isRequired,
  onChange: PropTypes.func,
  lang: PropTypes.string,
};

export default withUtils(styles)(ChangePassword);
