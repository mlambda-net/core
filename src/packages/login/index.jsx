export Register from './register';
export SignIn from './signin';
export Profile from './profile'
export ChangePassword from './changePassword'
export Recovery from './recovery'
