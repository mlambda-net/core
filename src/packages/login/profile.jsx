import React from 'react';
import PropTypes from 'prop-types';
import LocalizedStrings from 'react-localization';
import AccountCircleIcon from '@mui/icons-material/AccountCircle';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import Typography from '@mui/material/Typography';
import CardActions from '@mui/material/CardActions';
import { Box } from '@mui/material';
import { Option } from '@mlambda-net/core/common';
import withUtils from '@mlambda-net/core/utils/withUtils';

const language = new LocalizedStrings({
  en: {
    title: 'Profile',
  },
  es: {
    title: 'Perfil',
  },
});

const styles = (theme) => ({
  panel: {
    width: '100%',
    height: '100%',
  },

  title: {
    textAlign: 'center',
  },
  content: {
    height: 'calc(100% - 80px)',
  },
  actions: {
    display: 'flex',
    flexWrap: 'nowrap',
    justifyContent: 'flex-end',
  },
});

class Profile extends React.Component {
  constructor(props) {
    super(props);
    this.valid = this.validation;
    this.state = { open: false };
  }

  render() {
    language.setLanguage(this.props.lang);

    const Content = this.props.content;
    const Actions = this.props.actions;
    const { classes } = this.props;

    return (
      <Option
        icon={AccountCircleIcon}
        width={this.props.width}
        height={this.props.height}
        style={this.props.style}>
        <Card className={classes.panel}>
          <CardContent className={classes.content}>
            <Typography
              className={classes.title}
              color="textSecondary"
              variant="h5"
              gutterBottom>
              {language.title}
            </Typography>
            <Content />
          </CardContent>

          <CardActions className={classes.actions}>
            <Box>
              <Actions />
            </Box>
          </CardActions>
        </Card>
      </Option>
    );
  }
}

Profile.protoTypes = {
  className: '',
  classes: PropTypes.object.isRequired,
  user: PropTypes.object,
  lang: PropTypes.string,
  content: PropTypes.element,
  actions: PropTypes.element,
  width: PropTypes.string,
  height: PropTypes.string,
  style: PropTypes.string,
};

export default withUtils(styles)(Profile);
