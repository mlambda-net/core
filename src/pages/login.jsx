import React from 'react';
import Box from '@material-ui/core/Box';
import login from '../assets/img/login.png';
import { LeftImage } from '@mlambda-net/core/common';
import { SignIn } from '@mlambda-net/core/login';
import { withUtils } from '@mlambda-net/core/utils';
import Recovery from '@mlambda-net/core/login/recovery';
import { Action, State } from '@mlambda-net/core/common';

const styles = (themes) => ({
  root: {
    display: 'flex',
    backgroundRepeat: 'no-repeat',
    backgroundSize: 'cover',
    justifyContent: 'center',
    alignItems: 'center',
    height: '100%',
    width: '100%',
  },
  login: {},
});

class Login extends React.Component {
  constructor(props) {
    super(props);
    this.route = this.props.route;
    this.state = { actual: 'login' };
    this.onLogin = this.onLoginHandle.bind(this);
    this.onForgot = this.onForgotHandle.bind(this);
  }

  login(data) {
    this.route.to('store');
  }

  render() {
    const { classes } = this.props;

    return (
      <Box className={classes.root}>
        <Box width={'500px'}>
          <LeftImage img={<img src={login} width="100%" alt="img" />}>
            <Action actual={this.state.actual}>
              <State name="login">
                <SignIn
                  onForgot={() => this.setState({ actual: 'recovery' })}
                  onLogin={(user) => this.login(user)}
                />
              </State>
              <State name="recovery">
                <Recovery onClose={() => this.setState({ actual: 'login' })} />
              </State>
            </Action>
          </LeftImage>
        </Box>
      </Box>
    );
  }

  onLoginHandle(data) {
    console.log(data);
  }

  onForgotHandle() {}
}

export default withUtils(styles)(Login);
