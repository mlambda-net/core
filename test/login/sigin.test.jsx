import React from 'react';
import renderer from 'react-test-renderer';
import { expect, test } from "@jest/globals";
import { SignIn } from '@mlambda-net/core/login';
import Wrap from "../util";

test('Signing render', () => {
  const item = renderer.create(
    <Wrap>
      <SignIn title="Welcome"
              onLogin={() => console.log('login')}
              onForget={() => console.log('forget')} />
    </Wrap>
  );
  let tree = item.toJSON();
  expect(tree).toMatchSnapshot();
});
