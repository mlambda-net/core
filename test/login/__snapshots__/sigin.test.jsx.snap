// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Signing render 1`] = `
<div
  className="MuiBox-root css-8atqhb"
>
  <div
    className="MuiBox-root css-1l4w6pd"
  >
    <h5
      className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-lb0l52-MuiTypography-root"
    >
      Welcome
    </h5>
  </div>
  <div
    className="MuiBox-root css-8sg5vi"
  >
    <div
      className="MuiBox-root css-2rlxtj"
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorSecondary Mui-error MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-skv6of-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="email"
            id="email-label"
          >
            Email
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary Mui-error MuiInputBase-formControl css-1blo02g-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={true}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="email"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Email
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root css-2rlxtj"
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <div
          className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorSecondary Mui-error MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-skv6of-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="password"
            id="password-label"
          >
            Password
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary Mui-error MuiInputBase-formControl css-1blo02g-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={true}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Password
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root css-n823hc"
    >
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1aczlxi-MuiTypography-root-MuiLink-root"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <span
          className="MuiTypography-root MuiTypography-caption css-1man8r6-MuiTypography-root"
        >
          forgot password?
        </span>
      </a>
    </div>
    <div
      className="MuiBox-root css-1g72l7o"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium Mui-disabled MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-87ok9e-MuiButtonBase-root-MuiButton-root"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        login
      </button>
    </div>
  </div>
</div>
`;
